File: updated_code/code/python/1491-average-salary-excluding-the-minimum-and-maximum-salary.py
from typing import List

class Solution:
    """
    This class defines a method to calculate the average salary of employees
    excluding the minimum and maximum salary.
    """

    """
    This method takes a list of unique salaries as input and returns the
    average salary excluding the minimum and maximum salary.
    """

    def average(self, salary: List[int]) -> float:
        """
        Returns the average salary of employees excluding the minimum and
        maximum salary.

        Args:
        salary (List[int]): A list of unique salaries.

        Returns:
        float: The average salary excluding the minimum and maximum salary.
        """
        mini = min(salary)
        maxi = max(salary)
        return sum(i for i in salary if i != mini and i != maxi) / (len(salary) - 2.0)

File: updated_code/code/java/1491-average-salary-excluding-the-minimum-and-maximum-salary.java
package solution;

import java.util.Arrays;

public class Solution {
    
    public double average(int[] salary) {
        // Calculate sum, min and max
        int sum = 0;
        int min = Integer.MAX_VALUE;
        int max = Integer.MIN_VALUE;
        
        for (int i : salary) {
            sum += i;
            if (i < min) {
                min = i;
            } else if (i > max) {
                max = i;
            }
        }
        
        // Calculate average
        return (double) (sum - min - max) / (salary.length - 2);
    }    
}